lib/course_planner_web/controllers/offered_course_controller.ex

```diff
@@ -95,6 +126,46 @@ defmodule CoursePlannerWeb.OfferedCourseController do
     render(conn, "edit.html", offered_course: offered_course, changeset: changeset)
   end
 
+  def update(%{assigns: %{current_user: %{role: user_role, id: user_id}}} = conn,
+    %{"id" => id, "offered_course" => %{"syllabus" => syllabus}}) when user_role == "Teacher" do
```

Comment:
I think it's better if you match this inside a function extracted to `OfferedCourses`. It both cleans up the code on the controller level and leaves the modification logic to that module which is the responsible one.

Reply:
i am afraid considering the differences of the two update ( remember the update of students and classes and creating or removing of the attendance ) it might actually make it more complicated